src/kontrol/kdist/foundry.md

```diff
@@ -537,37 +537,42 @@ All cheat code calls which take place while `expectRevert` is active are ignored
       [priority(35)]
 ```
 
-The `#halt` production is used to examine the end of each call in KEVM.
-If the call depth of the current call is lower than the call depth of the `expectRevert` cheat code and the `<statusCode>` is not `EVMC_SUCCESS`, the `#checkRevertReason` will be used to compare the output of the call with the expect reason provided.
+We use the `#next[OP]` to identify OpCodes that can revert and insert a `#checkRevert` production used to examine the end of each call/create in KEVM.
+We also expect the call depth of the current call to be lower than the call depth of the `expectRevert`.
```

Comment:
Should it be "lower than or equal"?
```
requires CD <=Int ED
```

```suggestion
We also expect the call depth of the current call to be lower than or equal to the call depth of the `expectRevert`.
```

Reply:
It should be only `==`. @PetarMax suggested that I should write a test to check this behaviour:
 - expectRevert set with an expected message at depth 0.
 - there is a revert at depth 2 with some non-expected message which will be caught at depth 1
 - there is a revert at depth 1 with the expected message.

So I created a test like this that passes with Foundry
```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity =0.8.13;

import "forge-std/Test.sol";

contract TestExpectRevert is Test {
    function revertDepth2() public {
        revert ("This should be at depth 2");
    }
    function revertDepth1() public  {
        try this.revertDepth2()
        {} catch {}
        revert ("This should be at depth 1");
    }

    function test_expectRevert_inDepth() public {
        vm.expectRevert("This should be at depth 1");
        this.revertDepth1();
    }
}
```
![image](https://github.com/runtimeverification/kontrol/assets/16517508/1fe2faf6-9047-42de-8fa7-8c3cecf5b563)

Now, due to the `ED <=Int CD` (expected depth leq than the current call depth), the expect revert check would trigger before the `catch` in the bytecode and depth 1 and fail the `expectRevert`.
The point in having `ED ==Int CD` is that the expect revert check would only trigger at the same level as the call to the `expectRevert` cheat code invocation, which should be the foundry test.

Reply:
Thanks for the clarification! LGTM